walk.go

```diff
@@ -105,8 +108,15 @@ func walkFull(o interface{}, root map[string]interface{}, path llpath.Path, expa
 }
 
 // walkFull walks the given map[string]interface{} tree.
-func walkFullMap(m map[string]interface{}, root map[string]interface{}, p llpath.Path, expandPaths bool, wo walkObserver) (err error) {
-	for k, v := range m {
+func walkFullMap(mVal, rootVal reflect.Value, p llpath.Path, expandPaths bool, wo walkObserver) (err error) {
+	if mVal.Kind() != reflect.Map {
+		return fmt.Errorf("could not walk not map type for %s", mVal)
+	}
+
+	for _, kVal := range mVal.MapKeys() {
+		vVal := reflect.ValueOf(mVal.MapIndex(kVal).Interface())
```

Comment:
MapIndex already returns `reflect.Value`. Why get the interface and call ValueOf again.

Reply:
That is a bug, good catch.